src/app/pages/TopicPage/HierarchicalGrid/index.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import { string, shape, arrayOf, number, oneOfType, oneOf } from 'prop-types';
+import {
+  GEL_GROUP_1_SCREEN_WIDTH_MAX,
+  GEL_GROUP_2_SCREEN_WIDTH_MIN,
+  GEL_GROUP_2_SCREEN_WIDTH_MAX,
+  GEL_GROUP_3_SCREEN_WIDTH_MIN,
+  GEL_GROUP_3_SCREEN_WIDTH_MAX,
+  GEL_GROUP_4_SCREEN_WIDTH_MIN,
+} from '#psammead/gel-foundations/src/breakpoints';
+import Promo, { MEDIA_TYPES } from '#components/Promo';
+import { DESKTOP, TABLET, MOBILE, SMALL } from './dataStructures';
+
+const PromoList = styled.ul`
+  padding: 1.5rem;
+  margin: 0;
+  display: grid;
+  grid-gap: 10px;
+  grid-template-columns: repeat(4, 1fr);
+  @media (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
+    grid-template-columns: repeat(3, 1fr);
+  }
+  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
+    grid-template-columns: repeat(2, 1fr);
+  }
+`;
```

Comment:
I know we decided we had to use grid for this component but be good to see how this looks on Opera Mini and if we need to do some fallback styles for it